packages/create-angular/src/app/components/views/RiversView.component.ts

```diff
@@ -0,0 +1,363 @@
+      ></div>
+      <canvas
+        id="deck-canvas"
+        style="position: absolute; width: 100%; height: 100%"
+      ></canvas>
+      <app-layers
+        [layers]="layers()"
+        [layerVisibility]="layerVisibility()"
+        (layerVisibilityChanged)="layerVisibility.set($event)"
+      ></app-layers>
+      <app-card-collapsible title="Legend" class="legend">
+        <legend-entry-categorical
+          title="U.S. Rivers"
+          subtitle="By stream order"
+          [values]="legendValues()"
+          [getSwatchColor]="getSwatchColor"
+        ></legend-entry-categorical>
+      </app-card-collapsible>
+      @if (attributionHTML()) {
+        <aside class="map-footer" [innerHTML]="attributionHTML()"></aside>
+      }
+    </main>
+  `,
+})
+export class RiversViewComponent {
+  @ContentChild('maplibre-container', { static: true })
+  maplibreContainer?: HTMLDivElement;
+  @ContentChild('deck-canvas', { static: true }) deckCanvas?: HTMLCanvasElement;
+
+  private context = context;
+
+  constructor(private accessTokenService: AccessTokenService) {}
+
+  ngOnInit() {
+    this.map = new Map({
+      container: 'maplibre-container',
+      style: BASEMAP.DARK_MATTER,
+      interactive: false,
+    });
+
+    this.deck = new Deck({
+      canvas: 'deck-canvas',
+      initialViewState: INITIAL_VIEW_STATE,
+      controller: true,
+      layers: [],
+      onViewStateChange: ({ viewState }) => {
+        this.viewState.set(viewState);
+      },
+    });
+  }
+
+  /****************************************************************************
+   * DeckGL
+   */
+
+  private map: Map = null!;
+  private deck: Deck = null!;
+
+  private viewState = signal(INITIAL_VIEW_STATE);
+  // Debounce view state to avoid excessive re-renders during pan and zoom.
+  viewStateDebounced = debouncedSignal(this.viewState, 200);
+
+  attributionHTML = signal('');
+  droppingPercent = signal(0);
+
+  /****************************************************************************
+   * Sources (https://deck.gl/docs/api-reference/carto/data-sources)
+   */
+
+  data = computed(() => {
+    const accessToken = this.accessTokenService.accessToken();
+    if (!accessToken) {
+      // TODO: Prevent accessToken from being 'undefined' initially, after logging in
+      // to an app with OAuth enabled.
+      return new Promise(() => {}) as ReturnType<typeof vectorTilesetSource>;
+    }
+
+    return vectorTilesetSource({
+      accessToken,
+      apiBaseUrl: this.context.apiBaseUrl,
+      connectionName: CONNECTION_NAME,
+      tableName: TILESET_NAME,
+    });
+  });
+
+  roundedZoom = computed(() => Math.round(this.viewStateDebounced().zoom));
+
+  tilesLoaded = signal(false);
+
+  minStreamOrder = computed(() => getMinStreamOrder(this.viewState().zoom));
+
+  histogramTicks = computed(() => {
+    const ticks = [];
+    for (let i = 0; i <= MAX_STREAM_ORDER; i++) {
+      ticks.push(i);
+    }
+    return ticks;
+  });
```

Comment:
Totally optional, but a oneliner could be convenient here:

```suggestion
  histogramTicks = computed(() => Array.from({length: MAX_STREAM_ORDER}, (_, i) => i));
```

Reply:
Prettier set the formats for this code as 
```
histogramTicks = computed(() =>
  Array.from({ length: MAX_STREAM_ORDER }, (_, i) => i),
);
```